File: docs/current_docs/manuals/developer/snippets/documentation/go/object.go
package main

// The struct represents a single user of the system.
type MyModule struct {
	// The name of the user.
	Name string
	// The age of the user.
	Age int
}

func New(name string, age int) *MyModule {
	return &MyModule{
		Name: name,
		Age:  age,
	}
}

File: docs/current_docs/manuals/developer/snippets/documentation/typescript/object.ts
import { object, func } from "@dagger.io/dagger"

/**
 * The object represents a single user of the system.
 */
@object()
class MyModule {
  @func()
  name: string
  @func()
  age: number

  constructor(
    /**
     * The name of the user.
     */
    age: number,
    /**
     * The age of the user.
     */
    name: string
  ) {
    this.name = name
    this.age = age
  }
}
